homestar-runtime/src/event_handler/swarm_event.rs

```diff
@@ -341,23 +341,23 @@ async fn handle_swarm_event<THandlerErr: fmt::Debug + Send, DB: Database>(
             }
         },
         SwarmEvent::Behaviour(ComposedEvent::Mdns(mdns::Event::Discovered(list))) => {
-            for (peer_id, _multiaddr) in list {
+            for (peer_id, multiaddr) in list {
                 info!("mDNS discovered a new peer: {peer_id}");
                 event_handler
                     .swarm
                     .behaviour_mut()
-                    .gossipsub
-                    .add_explicit_peer(&peer_id);
+                    .kademlia
+                    .add_address(&peer_id, multiaddr);
             }
         }
         SwarmEvent::Behaviour(ComposedEvent::Mdns(mdns::Event::Expired(list))) => {
-            for (peer_id, _multiaddr) in list {
+            for (peer_id, multiaddr) in list {
                 info!("mDNS discover peer has expired: {peer_id}");
                 event_handler
                     .swarm
                     .behaviour_mut()
-                    .gossipsub
-                    .remove_explicit_peer(&peer_id);
+                    .kademlia
+                    .remove_address(&peer_id, &multiaddr);
```

Comment:
mdns will expire too quickly at the moment so this might remove nodes too eagerly.

Reply:
I'll wrap this around with something +1.

Reply:
Ah, it's dependent on the mdns ttl. Will configure that.